src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -184,44 +201,81 @@ private Future<Void> processInventoryOnlyWithHolding(CompositePoLine compPOL, Re
       .mapEmpty();
   }
 
+  private Future<List<Pair<String, String>>> deleteHoldings(String tenantId, List<JsonObject> holdings, RequestContext requestContext) {
+    if (CollectionUtils.isEmpty(holdings)) {
+      return Future.succeededFuture(List.of());
+    }
+    RequestContext newContext = createContextWithNewTenantId(requestContext, tenantId);
+    List<Future<Pair<String, String>>> deletedHoldingIds = new ArrayList<>(holdings.size());
+    holdings.forEach(holding -> {
+      String holdingId = holding.getString(ID);
+      String permanentLocationId = holding.getString(HOLDING_PERMANENT_LOCATION_ID);
+      if (holdingId != null) {
+        deletedHoldingIds.add(inventoryItemManager.getItemsByHoldingId(holdingId, newContext)
+          .compose(items -> {
+            if (items.isEmpty()) {
+              return inventoryHoldingManager.deleteHoldingById(holdingId, true, newContext)
+                .map(v -> Pair.of(holdingId, permanentLocationId));
+            }
+            return Future.succeededFuture();
+          }));
+      }
+    });
+    return collectResultsOnSuccess(deletedHoldingIds)
+      .map(resultDeletedHoldingVsLocationIds -> resultDeletedHoldingVsLocationIds.stream()
+        .filter(pair -> Objects.nonNull(pair) && Objects.nonNull(pair.getKey())).collect(toList()))
+      .map(resultDeletedHoldingVsLocationIds -> {
+        if (logger.isDebugEnabled()) {
+          String deletedIds = resultDeletedHoldingVsLocationIds.stream().map(Pair::getKey).collect(Collectors.joining(","));
+          logger.debug(String.format("Holdings were removed : %s", deletedIds));
+        }
+        return resultDeletedHoldingVsLocationIds;
+      });
+  }
+
+
   private Future<Void> processInventoryOnlyWithItems(CompositePoLine compPOL, RequestContext requestContext) {
     return deleteExpectedPieces(compPOL, requestContext)
       .compose(deletedPieces ->
-        inventoryItemManager.getItemsByPoLineIdsAndStatus(List.of(compPOL.getId()), ItemStatus.ON_ORDER.value(), requestContext)
-          .compose(onOrderItems -> {
-            if (CollectionUtils.isNotEmpty(onOrderItems)) {
-              List<String> itemIds = onOrderItems.stream().map(item -> item.getString(ID)).collect(toList());
-              return inventoryItemManager.deleteItems(itemIds, false, requestContext);
-            }
-            return Future.succeededFuture();
-          })
-          .onSuccess(v -> logger.info("Items and pieces deleted after Un-Open order with id: {}", compPOL.getId()))
-          .onFailure(e -> logger.error("Items and pieces deletion failed after Un-Open order with id: {}", compPOL.getId(), e))
+        GenericCompositeFuture.all(
```

Comment:
found one more thing, previous implementation was waiting when item deleted and only after that returned completed future, I think we need wait also to keep the same bahaviour, so need to add return before GenericCompositeFuture

Reply:
If we have 60 tenants and need update inventory objects in every of them, it can slow down process of un-opening the order, so probably we should remain as is with ignoring this comment, adding also @okolawole-ebsco to this discussion

Reply:
lets start with approach to wait until all items updated to not mask errors that can be produced by inventory, and after discussion will back to this and change if needed